packages/core/src/awsService/appBuilder/explorer/detectSamProjects.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { SamAppLocation } from './samProject'
+import { getLogger } from '../../../shared/logger/logger'
+import { getProjectRootUri } from '../../../shared/sam/utils'
+
+export async function detectSamProjects(): Promise<SamAppLocation[]> {
```

Comment:
why is this file in /explorer/ ? seems relevant in the /sam/ family of modules.

Reply:
It's used exclusively to populate the app builder view if a SAM project was detected in the workspace so I believe that's why we have it here. We can track moving this as a fast follow after launch.